data/alsa_drivers.rb

```diff
@@ -107,15 +106,27 @@ def params
 
   # find all sound drivers below the given path
   def self.find_all(path)
-    ret = []
-    lst = `find #{path} -type f -name 'snd-*.ko'`.split("\n").sort{|p1, p2| 
-	p1.split('/').last <=> p2.split('/').last
-    }
+    result = []
 
-    lst.each do |m|
-      ret << AlsaModule.new(m)
-    end
+    output = Command.output("find", path, "-type", "f", "-name", "snd-*.ko")
 
-    ret
+    drivers = output.split("\n").sort { |p1, p2| p1.split('/').last <=> p2.split('/').last }
+
+    drivers.map { |d| AlsaModule.new(d) }
+  end
+
+  # Auxiliary class to run a command and get its output
+  class Command
```

Comment:
Isn't it a bit overkill to have a class with only one class method? How about a simple method command_output() instead? If it's not generic enough for general use, it might even be private.

Reply:
Yes, my idea is to have that method as private. The problem is that the method is called from instance methods and also from class methods. So, if we define `command_output` as private class method, then it could not be called from instance methods.

Another possible option is to define `Command` as a module instead of a class.